File: Symfony/src/BGR/Serrano/ProductoBundle/Entity/Lote.php
<?php

namespace BGR\Serrano\ProductoBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use JMS\Serializer\Annotation\Type;
/**
 * Lote
 *
 * @ORM\Table()
 * @ORM\Entity(repositoryClass="BGR\Serrano\ProductoBundle\Entity\LoteRepository")
 */
class Lote
{
    /**
     * @var integer
     *
     * @Type("integer")
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     *
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    protected $id;

    /**
     * @var string
     *
     *
     * @Type("string")
     *
     * @ORM\Column(name="descripcion", type="string", length=255)
     */
    protected $descripcion;

    /**
     * @var \DateTime
     *
     * @Type("DateTime<'Y-m-d'>")
     *
     * @ORM\Column(name="fechaDeElaboracion", type="date")
     */
    protected $fecha_de_elaboracion;

    /**
     * @var \DateTime
     *
     * @Type("DateTime<'Y-m-d'>")
     *
     * @ORM\Column(name="fechaDeVencimiento", type="date")
     */
    protected $fecha_de_vencimiento;

    /**
     * @var double
     *
     * @Type("float")
     *
     * @ORM\Column(name="precio_compra", type="decimal")
     */
    protected $precio_compra;


    /**
     * Get id
     *
     * @return integer 
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set descripsion
     *
     * @param string $descripsion
     * @return Lote
     */
    public function setDescripcion($descripcion)
    {
        $this->descripcion = $descripcion;
    
        return $this;
    }

    /**
     * Get descripcion
     *
     * @return string 
     */
    public function getDescripcion()
    {
        return $this->descripcion;
    }

    /**
     * Set fechaDeElaboracion
     *
     * @param \DateTime $fechaDeElaboracion
     * @return Lote
     */
    public function setFechaDeElaboracion($fecha_de_elaboracion)
    {
        $this->fecha_de_elaboracion = $fecha_de_elaboracion;
    
        return $this;
    }

    /**
     * Get fechaDeElaboracion
     *
     * @return \DateTime 
     */
    public function getFechaDeElaboracion()
    {
        return $this->fecha_de_elaboracion;
    }

    /**
     * Set fechaDeVencimiento
     *
     * @param \DateTime $fechaDeVencimiento
     * @return Lote
     */
    public function setFechaDeVencimiento($fecha_de_vencimiento)
    {
        $this->$fecha_de_vencimiento = $fecha_de_vencimiento;
    
        return $this;
    }

    /**
     * Get fechaDeVencimiento
     *
     * @return \DateTime 
     */
    public function getFechaDeVencimiento()
    {
        return $this->$fecha_de_vencimiento;
    }

   


    /**
     * @ORM\ManyToOne(targetEntity="Producto") 
     * @ORM\JoinColumn(name="producto_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\Producto")
     *
     */
    protected $producto;

    /**
     * @ORM\ManyToOne(targetEntity="Proveedor")
     * @ORM\JoinColumn(name="proveedor_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\Proveedor")
     *
     */
    protected $proveedor;
    
    
    public function setProducto(Producto $producto)
    {
        $this->producto = $producto;
    }

    public function getProducto()
    {
        return $this->producto;
    }
	public function getPrecioCompra() {
		return $this->precio_compra;
	}
	public function setPrecioCompra( $precio_compra) {
		$this->precio_compra = $precio_compra;
		return $this;
	}
	public function getProveedor() {
		return $this->proveedor;
	}
	public function setProveedor($proveedor) {
		$this->proveedor = $proveedor;
		return $this;
	}
	
	



}

File: BGR/BGR-Data/src/main/java/ar/com/bgr/serrano/model/Lote.java
/**
 * 
 */
package ar.com.bgr.serrano.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

import ar.com.bgr.serrano.utils.DateJsonSerializer;

/**
 * 
 * Descripcion:
 * 
 * {}
 *
 * @author matias
 * 
 * @since 31/05/2014
 */
@Entity(name="Lote")
@Table(name="Lote")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Lote {

	@GeneratedValue
	@Id
	private Integer id;
	
	@Column( name = "producto_id")
	private int productoId;
	
	@Column( name = "descripcion")
	private String descripcion;
	
	@Column( name = "fechaDeElaboracion")
	private Date fechaDeElaboracion;

	@Column( name = "fechaDeVencimiento")
	private Date fechaDeVencimiento;

	@Column( name = "precio_compra")
	private double precio_compra;
	 
	@Column( name = "proveedor_id")
	private int proveedor_id;
	
	
	public Lote(){}

	/**
	 * @return the id
	 */
	public Integer getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(Integer id) {
		this.id = id;
	}

	/**
	 * @return the productoId
	 */
	public int getProductoId() {
		return productoId;
	}

	/**
	 * @param productoId the productoId to set
	 */
	public void setProductoId(int productoId) {
		this.productoId = productoId;
	}

	/**
	 * @return the descripcion
	 */
	public String getDescripcion() {
		return descripcion;
	}

	/**
	 * @param descripcion the descripcion to set
	 */
	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	/**
	 * @return the fechaDeElaboracion
	 */
	@JsonSerialize(using = DateJsonSerializer.class)
	public Date getFechaDeElaboracion() {
		return fechaDeElaboracion;
	}

	/**
	 * @param fechaDeElaboracion the fechaDeElaboracion to set
	 */
	public void setFechaDeElaboracion(Date fechaDeElaboracion) {
		this.fechaDeElaboracion = fechaDeElaboracion;
	}

	/**
	 * @return the fechaDeVencimiento
	 */
	@JsonSerialize(using = DateJsonSerializer.class)
	public Date getFechaDeVencimiento() {
		return fechaDeVencimiento;
	}

	/**
	 * @param fechaDeVencimiento the fechaDeVencimiento to set
	 */
	public void setFechaDeVencimiento(Date fechaDeVencimiento) {
		this.fechaDeVencimiento = fechaDeVencimiento;
	}

	/**
	 * @return the precio_compra
	 */
	public double getPrecio_compra() {
		return precio_compra;
	}

	/**
	 * @param precio_compra the precio_compra to set
	 */
	public void setPrecio_compra(double precio_compra) {
		this.precio_compra = precio_compra;
	}

	/**
	 * @return the proveedor_id
	 */
	public int getProveedor_id() {
		return proveedor_id;
	}

	/**
	 * @param proveedor_id the proveedor_id to set
	 */
	public void setProveedor_id(int proveedor_id) {
		this.proveedor_id = proveedor_id;
	}
	
	
	
}
